### PR TITLE
server: Create bookmarks file for experiments

### PR DESCRIPTION
**Pull request by PatrickTasse**
The bookmarks file can be used to store and retrieve markers associated with an experiment using platform APIs.

It can later be obtained for an opened experiment with: TmfTraceManager.getInstance().getTraceEditorFile(experiment)

### CODE REVIEW

**Review by MatthewKhouzam — APPROVED** [review]
Looks good to me!

File: trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ExperimentManagerService.java [hunk 1]
```diff
@@ -495,6 +498,24 @@ private static void createSupplementaryFolder(IResource experimentResource) {
...
+    private static IFile createBookmarksFile(IResource resource) {
+        IFile file = ((IFolder) resource).getFile(resource.getName() + '_');
+        try {
+            if (!file.exists()) {
+                final IFile bookmarksFile = ((IFolder) resource.getParent()).getFile(BOOKMARKS_HIDDEN_FILE);
+                if (!bookmarksFile.exists()) {
+                    final InputStream source = new ByteArrayInputStream(new byte[0]);
+                    bookmarksFile.create(source, IResource.FORCE | IResource.HIDDEN, new NullProgressMonitor());
+                }
+                file.createLink(bookmarksFile.getLocation(), IResource.REPLACE | IResource.HIDDEN, new NullProgressMonitor());
```
[the remark this answers is not in the document]
Comment (PatrickTasse) [reply trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ExperimentManagerService.java:510]: According to createLink() Javadoc, only the mentioned flags have an effect while other flags are ignored, and FORCE is not mentioned.

[the remark this answers is not in the document]
Comment (MatthewKhouzam) [reply trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ExperimentManagerService.java:510]: Should we update the other one then TmfCommonProjectElement.java:394

[the remark this answers is not in the document]
Comment (PatrickTasse) [reply trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ExperimentManagerService.java:510]: It's handled in create() but ignored in createLink().

[the remark this answers is not in the document]
Comment (MatthewKhouzam) [reply trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ExperimentManagerService.java:510]: +1 still applies.

**Pull request merged**